Libs/Professions/professions.lua

```diff
@@ -0,0 +1,137 @@
+
+Professions = {
+    UIOptions = {
+        header = {
+            order = 0,
+            type = "header",
+            name = "Professions",
+        },
+        description = {
+            order = 1,
+            type = "description",
+            name = function() return Professions:Description() end,
+        },
+        selectprofession = {
+            order = 2,
+            type = "select",
+            name = "Select Profession",
+            values = {
+                Alchemy = "Alchemy",
+                Blacksmithing = "Blacksmithing",
+                Enchanting = "Enchanting",
+                Leatherworking = "Leatherworking",
+                Tailoring = "Tailoring"
+            },
+            get = function(info)
+                return selectedProfession;
+            end,
+            set = function(info, value)
+                Professions:updateOptions(value);
+            end,
+        },
+        selectsubprof = {
+            order = 3,
+            type = "select",
+            name = "Category",
+            values = subProfessionValues;
+            get = function(info)
+                return selectedSubProfession;
+            end,
+            set = function(info, value)
+                Professions:updateSubOptions(value);
+            end,
+        },
+        professors = {
+            order = 4,
+            type = "description",
+            name = function() return Professions:updateProfessors() end;
+        }
+    }
+}
+
+function Professions:updateOptions(value)
+    selectedProfession = value;
+    selectedSubProfession = "";
+    count = #subProfessionValues
+    for i=0, count do subProfessionValues[i]=nil end
+    if value == "Alchemy" then
+        table.insert(subProfessionValues, "Flasks");
+        table.insert(subProfessionValues, "Elixirs");
+        table.insert(subProfessionValues, "Resist Potions");
+        table.insert(subProfessionValues, "Miscellaneous");
+    elseif value == "Blacksmithing" then
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Weapon");
+        table.insert(subProfessionValues, "Miscellaneous");
+    elseif value == "Enchanting" then
+        table.insert(subProfessionValues, "Boots");
+        table.insert(subProfessionValues, "Bracer");
+        table.insert(subProfessionValues, "Chest");
+        table.insert(subProfessionValues, "Cloak");
+        table.insert(subProfessionValues, "Gloves");
+        table.insert(subProfessionValues, "Shield");
+        table.insert(subProfessionValues, "Weapon");
+        table.insert(subProfessionValues, "Oils");
+    elseif value == "Leatherworking" then
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Cloak");
+    elseif value == "Tailoring" then
+        table.insert(subProfessionValues, "Bags");
+        table.insert(subProfessionValues, "Armor");
+        table.insert(subProfessionValues, "Cloak");
+    end
+end
```

Comment:
ok, This is valid but not optimum. What happen if we want to add a new profession like `Cooking` ? we will have to add another condition like 
```
    elseif value == "Cooking" then
        table.insert(subProfessionValues, "Food1");
        table.insert(subProfessionValues, "Food2");
        table.insert(subProfessionValues, "Food3");
    end
```
and what if we want another one like `Inscription` _(for future wow expansions)_ the problem is that we are ending up with a huge method and a lot of `elsif` condition ,  that doesn't sound good.

Also I think the responsable for knowing wich subprofessions has alchemy is **Alchemy.lua**.  The responsable for knowing which subprofessions has **Enchanting** is **Enchanting.lua** and go on.

We know that the variable **value** is a string representation of< Alchemy, Tailoring , etc>.  And we know we can access the respective object using `_G[tablename]:` we could probably do something like this for getting the `subProfessionValues`. ... ---> ` _G[value].subProfessions`

```
function Professions:updateOptions(value)
    selectedProfession = value;
    selectedSubProfession = _G[value].subProfessions
end
```
or just return the subProfesions
```

function Professions:subProfessions(value)
    return _G[value].subProfessions
end
```

This means that **Alchemy.lua** **Tailoring.lua** **Leatherworking.lua** etc... must have an attribute called "**subProfessions**" like this
```
Tailoring = {
    subProfessions = {  BaG1="BAG1", Boots1 = "Boots1",etc }
    bags = {
.
.
```


And if we add a new professions we just need to add the name of that profession to the list of professions **above** like this
```
Professions = {
    UIOptions = {
      .
      .
      selectprofession = {
            order = 2,
            type = "select",
            name = "Select Profession",
            values = {
                Alchemy = "Alchemy",
                Blacksmithing = "Blacksmithing",
                Enchanting = "Enchanting",
                Leatherworking = "Leatherworking",
                Tailoring = "Tailoring",
             Cooking= "Cooking",
             Inscription= "Inscription",
             SoftwareDev= "SoftwareDev",
            },
```

Reply:
This is called Duck Typing ,  look https://www.youtube.com/watch?v=CuK0g8OFzwo&ab_channel=Telusko

Reply:
Ya the bummer with lua is lack of case switch statements. You make a good point I could populate the "subProfessions" list in the actual profession and just get it from there. I wish lua had pointers so I could just update those accordingly but I guess this will do :upside_down_face: